codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -315,8 +315,9 @@ private void increaseInitialConnectionWindow(int deltaBytes) throws Http2Excepti
 
     final boolean consumeBytes(int streamId, int bytes) throws Http2Exception {
         Http2Stream stream = connection().stream(streamId);
-        // upgraded requests are ineligible for stream control
-        if (streamId == Http2CodecUtil.HTTP_UPGRADE_STREAM_ID) {
+        // Upgraded requests are ineligible for stream control. We add the null check
+        // in case the stream has been deregistered.
+        if (stream != null && streamId == Http2CodecUtil.HTTP_UPGRADE_STREAM_ID) {
```

Comment:
Actually should we just return early if `stream == null` as otherwise we will pass `null` to the flow-controller ?

Reply:
I think it is technically consistent with the methods contract to return early, but in these situations I tend to let the flow-controller make the final decision. The flow-controller should make the right move and it gives any other proxies a chance to observe the event, although the value of that is debatable.